### ISSUE TITLE
Python 2.7.x

### ISSUE THREAD

**Issue by hairesis** [rumpkernel/rumprun-packages#84]
Is there a reason Python2.7.x has not been targeted? Would be nice to have it as a supported language.
I've tried to compile 2.7.11 using the actual 3.5 package recipe but it does not work.
`mv: cannot stat ‘build/Programs/_freeze_importlib’: No such file or directory
Makefile:48: recipe for target 'build/Programs/host_freeze_importlib' failed
make: *** [build/Programs/host_freeze_importlib] Error 1
`
Did someone already tried to get it built?

**Comment by anttikantee** [comment]
As far as I know, the only reason for lack of python 2.7 support is that nobody has looked into it and made it happen.  I'd surely like to see python2 supported.

**Comment by reuben-sutton** [comment]
I'd also love to see Python 2.7 support. If you don't want to do it @hairesis let me know and I'll have a look at adding it this weekend.

**Comment by hairesis** [comment]
Well, I am new to the project, but I can certainly try to work on it at least as an exercise, than we can see how it goes.

**Comment by anttikantee** [comment]
As long as everyone doesn't sit around and expect the other person to do it, it's all good ;-)

There are also plenty of other interesting languages which would be nice to support, e.g. ponylang.org.

@hairesis if you run into problems, drop a mail on the list or hang around on irc.

**Comment by chrbaeu** [comment]
I also would like Python 2.7 support. Has anyone since then found a way to use it?

**Comment by rday** [comment]
2.7 wouldn't be impossible to implement, but I would suggest trying to get
your application up on the 3.5 that is currently supported.

The reason I suggest this is because most third party modules today support
3.5. Anything that is custom enough to not be supported on 3.5 may require
a fairly custom 2.7 build anyway. This wouldn't be supported "out of the
box".

Are there any specific issues you've hit while trying to use the 3.5
version?

**Comment by anttikantee** [comment]
On 02/09/16 11:25, Ryan Day wrote:

> 2.7 wouldn't be impossible to implement, but I would suggest trying to get
> your application up on the 3.5 that is currently supported.
> 
> The reason I suggest this is because most third party modules today support
> 3.5. Anything that is custom enough to not be supported on 3.5 may require
> a fairly custom 2.7 build anyway. This wouldn't be supported "out of the
> box".
> 
> Are there any specific issues you've hit while trying to use the 3.5
> version?

Speaking just for myself, I have a python script I've been writing for 
several years now.  It works only with python2, at least the syntax of 
"print" is incompatible with python3.  While I could convert the script 
to python3 syntax (possibly automatically), at least as long as "python" 
is python2 on most regular systems, it seems rather pointless to do the 
extra work.  I'll just wait for python4 instead ;)

While I don't want to run that script under Rumprun, I can see why 
python2 support would be desirable for others.

**Comment by Incognito** [comment]
From my perspective (and my day-job is writing python now), python 2 is still widely used and new software is developed with it.

**Comment by chrbaeu** [comment]
Yes that’s true python 2 is still widely used and there is a lot of existing code. Support for it would be nice. A porting on python 3 is usually too much work without a special reason for it. But perhaps in this case it is such a reason and it makes more sense to port it on python 3.

**Comment by rday** [comment]
All my production applications (with the exception of one) are also in Python 2.7. So I would have a hard time arguing against it :)

I've taken an initial look at the build system for 2.7.12 and the differences aren't too extreme. I'm putting together a build system and hope to have something up soon. I'll keep this thread updated.
